app/models/vacancy.rb

```diff
@@ -144,7 +144,12 @@ def self.remove_vacancies_that_expired_yesterday!
         religious_character: school.gias_data['ReligiousCharacter (name)'],
         region: school.region&.name,
         school_type: school.school_type&.label&.singularize,
-        town: school.town }
+        town: school.town } if school.present?
+    end
+
+    attribute :school_group do
+      school_group = self.school_group
+      { uid: school_group&.uid }
```

Comment:
is defining the method `uid` in scope?

Reply:
We don't need a `uid` method?

Reply:
Nonetheless, it looks like you are using it

Reply:
I know `&.` is the uber concise way to do this, but it does make the code harder to read.  Can we please go with `school_group.try(:uid)` instead?  It looks less like perl-ish line noise.

Reply:
Also, I'm not entirely clear on what this block is supposed to be doing anymore.  Is it assigning the school_group object to the Algolia index, or is only yielding `{ uid: <some uid> }` presumably as JSON?

Reply:
@tatyree the algolia block here was a placeholder to indicate that the school_group will likely need to be indexed similarly to the school. It's not in the scope of this ticket, so I will remove it

Reply:
@david-mears-dfe It's a field on the record, not a method